osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/DlmsConnectionMessageProcessor.java

```diff
@@ -52,11 +53,19 @@
     @Autowired
     protected DlmsDeviceRepository deviceRepository;
 
-    protected DlmsConnectionManager createConnectionForDevice(final DlmsDevice device,
-            final MessageMetadata messageMetadata) throws OsgpException {
+    @Autowired
+    protected ThrottlingService throttlingService;
+    
+    public DlmsConnectionManager createConnectionForDevice(final DlmsDevice device,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to throw at most one checked exception instead of: org.opensmartgridplatform.shared.exceptionhandling.OsgpException, java.lang.InterruptedException [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1160)